### DISCUSSION TITLE
Conditional wire:navigate in blade component

### DISCUSSION THREAD

**Discussion by akunbeben** [livewire/livewire#6733] [1. Help] [answered]
I have a blade component that uses the `wire:navigate`, in the Livewire component we can use the `$this->redirect('/url', redirect: false)` to determine whether using `wire:navigate` or not, how can I achieve this in a blade component (conveniently)?

I did achieve this using "if else" statement. But, I want something like `wire:navigate="false"`? So, we can just pass the navigate value from the component props.

**Comment by N1ebieski** [comment]
Maybe use a [directive](https://laravel.com/docs/10.x/blade#extending-blade):

```php
Blade::directive('wireNavigate', function (bool $enable = true) {
    return $enable ? "wire:navigate" : "";
});
```

```php
<a href="route('something')" @wireNavigate($prop)>link</a>
```

**Comment by joshhanley** [comment, marked as answer]
@akunbeben you can just pass the props down when you need it.

If you have this component
```blade
// my-link.blade.php
<a {{ $attributes }} >{{ $slot }}</a>
```

And call it like this
```blade
<x-my-link wire:navigate href="/somewhere">Somewhere</a>
```

It will render
```blade
<a wire:navigate href="/somewhere">Somewhere</a>
```

If you call it without `wire:navigate`
```blade
<x-my-link href="/somewhere">Somewhere</a>
```

It will render
```blade
<a wire:navigate href="/somewhere">Somewhere</a>
```
Hope this helps!

**Reply by akunbeben** [reply]
Thanks, Josh!
